web/assets/js/components/gamecontrols.ts

```diff
@@ -1,185 +1,206 @@
+                this.DOM.timer.innerHTML = this.timer;
+                gsap.set(this.DOM.timer, {autoAlpha: 0, display: "none"});
+            });
+        }
+    }
+
+    async _clickOnConfirm() {
+        const actions: Actions = await Actions.getInstance()
+
+        if (this.action === "resign") {
+            await actions.requestResign(""); // TODO @Alexis add game ID
+            this.call("goTo", ["/"], "router");
+        }
+        if (this.action === "draw") {
+            if (this.pendingDraw) {
+                clearInterval(this.pendingDraw);
+                this.timer = 9;
+                this.pendingDraw = null;
+            } else {
+                //TODO: wait screen
+                const game: Game = await actions.requestDraw(""); // TODO @Alexis add game ID
+                // TODO @Alexis, you need to check the game state here,
+                // or set up a poller for checking if a draw happens (if you don't validate the state on each move).
+                //
+                // if (game.state === GameState.DRAWN_BY_AGREEMENT) {
+                //     this.call("engine", [false, "draw"], "gameboard");
+                // }
+                console.log(game)
+            }
+        }
+
+        this.validationTL.reverse();
+
+        this[this.action === "resign" ? "disableCtr1TL" : "disableCtr0TL"].reverse();
+        this[this.action === "resign" ? "swithCtr0TL" : "swithCtr1TL"].reverse();
+
+        //TODO: link to game component to draw or quit
+        this.action = "void";
+    }
+
+    async _declineOffer() {
+        const actions: Actions = await Actions.getInstance()
+
+        await actions.declineDraw(""); // TODO @Alexis add game ID
         clearInterval(this.pendingDraw);
         this.timer = 9;
         this.pendingDraw = null;
-      } else {
-        //TODO: wait screen
-        const isAccepted = await Action.requestDraw();
-        if (isAccepted) {
-          this.call("engine", [false, "draw"], "gameboard");
-        }
-      }
     }
 
-    this.validationTL.reverse();
+    _getDrawProposition() {
+        gsap.set(this.DOM.timer, {autoAlpha: 1, display: "inline-block"});
 
-    this[this.action === "resign" ? "disableCtr1TL" : "disableCtr0TL"].reverse();
-    this[this.action === "resign" ? "swithCtr0TL" : "swithCtr1TL"].reverse();
+        this.pendingDraw = setInterval(() => {
+            this.timer--;
+            this.DOM.timer.innerHTML = this.timer;
 
-    //TODO: link to game component to draw or quit
-    this.action = "void";
-  }
+            if (this.timer <= 0) {
+                this._clickOnCtr("draw", true);
+                this.timer = 9;
+            }
+        }, 1000);
+        this._clickOnCtr("offer", false);
+        console.log("propal received");
 
-  _declineOffer() {
-    Action.declineDraw();
-    clearInterval(this.pendingDraw);
-    this.timer = 9;
-    this.pendingDraw = null;
-  }
+        // TODO @Alexis is something supposed to happen here with Actions?
```

Comment:
The Action.listenDraw() method has disappeared, so I guess we need to recreate a `listenDraw` polling Action method to check if a draw has been requested by the rival.

Reply:
Added support for it here:
cabbcc8d07159d472fc94d05db4f156059aac2cc